moveit_core/src/collision_detection_fcl/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+target_sources(moveit_core PRIVATE
+  collision_common.cpp
+  collision_env_fcl.cpp
+  collision_detector_fcl_plugin_loader.cpp
+)
+
+if(BUILD_TESTING)
+  if(WIN32)
+    # set(append_library_dirs "$<TARGET_FILE_DIR:${PROJECT_NAME}>;$<TARGET_FILE_DIR:${PROJECT_NAME}_TestPlugins1>")
+  else()
+    set(append_library_dirs "${CMAKE_CURRENT_BINARY_DIR};${CMAKE_CURRENT_BINARY_DIR}/../collision_detection;${CMAKE_CURRENT_BINARY_DIR}/../robot_state;${CMAKE_CURRENT_BINARY_DIR}/../robot_model;${CMAKE_CURRENT_BINARY_DIR}/../utils")
+  endif()
+
+  # TODO: remove if transition to gtest's new API TYPED_TEST_SUITE_P is finished
+  if(CMAKE_CXX_COMPILER_ID MATCHES "(GNU|Clang)")
+    add_compile_options(-Wno-deprecated-declarations)
+  endif()
+
+  ament_add_gtest(test_fcl_collision_env test/test_fcl_env.cpp)
+  target_link_libraries(test_fcl_collision_env moveit::core)
```

Comment:
Doesn't this change the behavior of the test? The old cmake didn't link against the plugin code so it implicitly tested for valid plugin registration and loading. This version might omit this step, no?

Reply:
That's a good point. I think it's reasonable to keep the plugins as separate targets. I'll have to think about that more.

Reply:
It is best practice to keep plugin lib instantiations in a separate standalone library and I would strongly propose you keep it that way. We had a similar case [in MTC last year](https://github.com/ros-planning/moveit_task_constructor/pull/271#issuecomment-850671203) (that I brought up because of the infamous memory leaks) and ended up splitting the plugins off again.

Personally my main argument is that unless strictly necessary, libraries should not execute code on so-load, unless strictly needed for *every use of the library*. But that's exactly what the PLUGINLIB/CLASSLOADER macros do and not every use of `moveit::core` requires plugins. You have no idea how often boost libraries broke my neck as a linux distribution maintainer because of some random unnecessary initialization it does on load.

Reply:
Viable options I see to move the plugins out:
- new moveit_plugins subfolders (actually new packages) for all
- a separate `moveit::core_plugins` library for the three of them
- add `moveit::collision_plugins` and split the butterworth filter out into moveit_plugins. fcl/bullet are definitely close to moveit::core, but the signal smoother is only used in moveit_servo. On the other hand moveit_core does not use the plugins but directly relies on the classes in most cases.

I slightly prefer the last option.